model/Index/Listener/ResourceOperationAdapter.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2022 (original work) Open Assessment Technologies SA.
+ */
+
+namespace oat\taoAdvancedSearch\model\Index\Listener;
+
+use core_kernel_classes_Resource;
+use oat\generis\model\data\event\ResourceUpdated;
+use oat\tao\model\TaoOntology;
+use oat\taoTests\models\event\TestUpdatedEvent;
+
+/**
+ * This service receives resource instances and triggers an event though the
+ * provided event listener matching the called method and resource type.
+ */
+class ResourceOperationAdapter
```

Comment:
I am not sure this is an Adapter. What is the common interface they are trying to adapt to? Also, what is a `resource operation`? 

https://en.wikipedia.org/wiki/Adapter_pattern

I am not sure we should factory events inside this project just to make them compatible with listeners. We are increasing too much the complexity IMO

Reply:
It is not ideal, but it is needed to be able to handle the resources that are to be indexed by `SyncResourceResultIndexer` without duplicating here the same logic used for events.

This way we can have the listeners mapped by event type as a configuration parameter in the DI container and use here the same mapping to handlers. Having a single point to handle both (event listeners & calls to `SyncResourceResultIndexer`) was one of the goals of this PR.

However, maybe we can also have the map of types to events as a config parameter too, and/or add an additional method to `AbstractEventHandler` that receives a resource instead of an event (and use the methods already provided by child classes to convert the received events to resource instances).

Reply:
We may also move this logic to `SyncResourceResultIndexer` and remove this class (moving it into a new class was done in order for it to be reusable in other classes if needed).

Reply:
Well, I've been thinking a bit about this and, as this class is not really adapting an interface for an object itself (as you already pointed) but just the interactions between them, maybe `ResourceOperationMediator` is a better name.

Reply:
Done :ok_hand: f447c53e919223c698fddc949ec6eab9746468b5